lucene/core/src/java/org/apache/lucene/search/ScoreAndDoc.java

```diff
@@ -32,4 +33,9 @@ public int docID() {
   public float score() {
     return score;
   }
+
+  @Override
+  public float smoothingScore(int docId) throws IOException {
```

Comment:
Hmm, why not add a default method in the `Scorable` interface to `return 0`?  Then we don't have to add this default method in all these subclasses?

Reply:
I agree that this makes more sense ;-)  I have added the default implementation of smoothingScore in Scorable and reverted my changes to add the smoothingScore method to all the unnecessary extending classes.